src/engine/bank3.asm

```diff
@@ -1380,42 +1586,1346 @@ RST20: ; cc42 (3:4c42)
+	ld a, [wd3b6]
+	ld [wd3aa], a
+	jr Func_ce3a
+
+Func_ce52: ; ce52 (3:4e52)
+	ld a, [wd3aa]
+	push af
+	ld a, [wd3ab]
+	push af
+	ld a, [$d696]
+asm_ce5d
+	ld [wd3ab], a
+	call Func_39c3
+	call Func_ce3a
+	pop af
+	ld [wd3ab], a
+	pop af
+	ld [wd3aa], a
+	ret
+
+Func_ce6f: ; ce6f (3:4e6f)
+	ld a, [wd3aa]
+	push af
+	ld a, [wd3ab]
+	push af
+	ld a, c
+	push af
+	call Func_cc9a
+	push bc
+	call Func_cc60
+	pop bc
+	pop af
+	jr asm_ce5d
+
+Func_ce84: ; ce84 (3:4e84)
+	call Func_c135
+	jp Func_cc60
+
+Func_ce8a: ; ce8a (3:4e8a)
+	xor a
+	ld [wd117], a
+	push bc
+	call Func_c2a3
+	pop bc
+	push bc
+	ld a, c
+	farcall BoosterPack_1031b
+	ld a, 1
+	ld [wd117], a
+	pop bc
+	ld a, b
+	cp $ff
+	jr z, .asm_ceb4
+	farcall BoosterPack_1031b
+	call Func_cc9e
+	ld a, c
+	cp $ff
+	jr z, .asm_ceb4
+	farcall BoosterPack_1031b
+.asm_ceb4
+	call Func_c2d4
+	jp Func_cc68
+
+Func_ceba: ; ceba (3:4eba)
+	xor a
+	ld [wd117], a
+	call Func_c2a3
+	ld hl, $4edd
+.asm_cec4
+	ld a, [hl]
+	cp $ff
+	jr z, .asm_ced7
+	push hl
+	farcall BoosterPack_1031b
+	ld a, $1
+	ld [wd117], a
+	pop hl
+	inc hl
+	jr .asm_cec4
+.asm_ced7
+	call Func_c2d4
+	jp Func_cc60
+; 0xcedd
+
+INCBIN "baserom.gbc",$cedd,$cee2 - $cedd
+
+Func_cee2: ; cee2 (3:4ee2)
+	call Func_c2a3
+	ld a, c
+	cp $ff
+	jr z, .asm_cf09
+	or a
+	jr nz, .asm_cef0
+	ld a, [$d697]
+
+.asm_cef0
+	push af
+	farcall Func_10000
+	farcall Func_10031
+	pop af
+	bank1call Func_7594
+	call Func_c1a4
+	call DoFrameIfLCDEnabled
+	call Func_c2d4
+	jp Func_cc64
+
+.asm_cf09
+	xor a
+	jr .asm_cef0
+
+Func_cf0c: ; cf0c (3:4f0c)
+	ld a, c
+	call Func_1ce1
+	jr asm_cf16
+
+Func_cf12: ; cf12 (3:4f12)
+	ld a, c
+	call $1d1d
+
+asm_cf16
+	or a
+	jr nz, asm_cf1f
+
+asm_cf19
+	call Func_ccb9
+	jp Func_cc68
+
+asm_cf1f
+	call Func_ccb3
+	call Func_cc9a
+	jr z, asm_cf2a
+	jp Func_cc8b
+
+asm_cf2a
+	jp Func_cc68
+
+Func_cf2d: ; cf2d (3:4f2d)
+	push bc
+	call Func_cc60
+	pop bc
+	call Func_1caa
+	ld a, h
+	cp b
+	jr nz, .asm_cf3b
+	ld a, l
+	cp c
+
+.asm_cf3b
+	jr nc, asm_cf1f
+	jr asm_cf19
+
+Func_cf3f: ; cf3f (3:4f3f)
+	ld a, c
+	or a
+	jr nz, .asm_cf46
+	ld a, [$d697]
+
+.asm_cf46
+	call AddCardToCollection
+	jp Func_cc64
+
+Func_cf4c: ; cf4c (3:4f4c)
+	ld a, c
+	call Func_1d91
+	jp Func_cc64
+
+Func_cf53: ; cf53 (3:4f53)
+	ld c, $1
+	ld b, $0
+.asm_cf57
+	ld a, c
+	call Func_1d1d
+	add b
+	ld b, a
+	inc c
+	ld a, c
+	cp $8
+	jr c, .asm_cf57
+	ld a, b
+	or a
+	jr nz, Func_cf6d
+Func_cf67: ; cf67 (3:4f67)
+	call Func_ccb9
+	jp Func_cc78
+
+Func_cf6d: ; cf6d (3:4f6d)
+	call Func_ccb3
+	call Func_cc96
+	jr z, .asm_cf78
+	jp Func_cc8b
+
+.asm_cf78
+	jp Func_cc78
+
+Func_cf7b: ; cf7b (3:4f7b)
+	ld c, $1
+.asm_cf7d
+	push bc
+	ld a, c
+	call Func_1d1d
+	jr c, .asm_cf8c
+	ld b, a
+.asm_cf85
+	ld a, c
+	call Func_1d91
+	dec b
+	jr nz, .asm_cf85
+
+.asm_cf8c
+	pop bc
+	inc c
+	ld a, c
+	cp $8
+	jr c, .asm_cf7d
+	jp Func_cc60
+
+; This function doesn't look like a valid function, but it's pointed to in the table.
+Func_cf96: ; cf96 (3:4f96)
```

Comment:
This is pointed to on the long pointer table in bank 4. It looks really wacky though. I'm thinking ca69 might do some weird stack manip I don't know about or something. I don't see any reason why there would ever ever be `ld [$0138], sp` in legitimate code.